tests/test_nominatim_gazetteer.py

```
@@ -0,0 +1,215 @@
+        gz = ng.NominatimGazetteer(cache_size=4)
+        gz.lookup("Birmingham", country_hint="US")
+        gz.lookup("Birmingham", country_hint="US")
+        assert fake_client.geocode.call_count == 1
+
+
+# ---------------------------------------------------------------------------
+# Factory
+# ---------------------------------------------------------------------------
+
+class TestFactory:
+    def test_prefer_nominatim(self, fake_client):
```

Ruff (0.15.12)

[warning] Unused method argument: `fake_client`

(ARG002)